File: XDUploadClient/src/main/kotlin/com/xdupload/Environment.kt
package com.xdupload

import java.io.*
import java.util.*

/**
 * Created by xingda on 16-8-24.
 */
internal object Environment {

    val sPackageName: String
    val sReleaseNote: String
    val sCdnDomain: String
    val sEndpoint: String
    val sAccessKeyId: String
    val sAccessKeySecret: String
    val sBucketName: String
    val sPathPrefix: String

    init {
        val properties = Properties()
        properties.load(StringReader(Utils.readPropertiesAsString()))
        sPackageName = properties.getProperty("packageName")
        sReleaseNote = properties.getProperty("releaseNote")
        sCdnDomain = properties.getProperty("cdnDomain")
        sEndpoint = properties.getProperty("endpoint")
        sAccessKeyId = properties.getProperty("accessKeyId")
        sAccessKeySecret = properties.getProperty("accessKeySecret")
        sBucketName = properties.getProperty("bucketName")
        sPathPrefix = properties.getProperty("pathPrefix")
    }
}

File: XdUploadClient/src/main/scala/com/xdandroid/xduploadclient/Environment.scala
package com.xdandroid.xduploadclient

import java.io.StringReader
import java.util.Properties


/**
  * Created by xingda on 16-11-4.
  */
object Environment {

  val properties = new Properties()
  properties.load(new StringReader(Utils.readPropertiesAsString))

  val sPackageName = properties.getProperty("packageName")
  val sReleaseNote = properties.getProperty("releaseNote")
  val sCdnDomain = properties.getProperty("cdnDomain")
  val sEndpoint = properties.getProperty("endpoint")
  val sAccessKeyId = properties.getProperty("accessKeyId")
  val sAccessKeySecret = properties.getProperty("accessKeySecret")
  val sBucketName = properties.getProperty("bucketName")
  val sPathPrefix = properties.getProperty("pathPrefix")
}
